src/gui/sdlmain.cpp

```diff
@@ -3850,6 +3842,22 @@ bool GFX_Events()
 			continue;
 		}
 		switch (event.type) {
+		case SDL_JOYDEVICEADDED: {
+			SDL_JoyDeviceEvent* added_event =
+			        reinterpret_cast<SDL_JoyDeviceEvent*>(&event);
+			LOG_WARNING("MAPPER: Joystick device was just added, index: '%d'",
```

Comment:
Comments about both the SDL_JOYDEVICEADDED and SDL_JOYDEVICEREMOVED log messages:

- Should be info level, not warn.
- Should be ", instance ID %d" for both, but these instances IDs tell absolutely nothing to the user. Would it be possible to look up the human-readable name of the device as it reports itself to the OS and log that name instead?
- I'm not quite sure how the differentiation between "joysticks" and "game controllers" work in SDL. Don't we need to handle both (see [this event](https://github.com/libsdl-org/SDL/blob/main/include/SDL3/SDL_events.h#L645-L665)? In your tests you were using Xbox game controllers, so I'm a bit confused why these report themselves as "joysticks", not "game controllers". What about the extra buttons on game controllers like the Xbox 360; those are fully mappable in Staging. So wouldn't it be better if we only dealt with "game controllers" conceptually?
- So, I suggest to log something like this instead: `MAPPER: Game controller Xbox 360 has been plugged in." (or whatever the human-readable name of the controller is). And then `MAPPER: Game controller Xbox 360 has been disconnected."

Reply:
> Should be info level, not warn.

Fixed

> Should be ", instance ID %d" for both, but these instances IDs tell absolutely nothing to the user. Would it be possible to look up the human-readable name of the device as it reports itself to the OS and log that name instead?

According to the SDL documentation I found, it's index when added and instance id when removed. However I agree that either way this is not helpful to the user. I will have to do a bit of research to figure out how the get the human readable controller name.

> I'm not quite sure how the differentiation between "joysticks" and "game controllers" work in SDL. Don't we need to handle both (see [this event](https://github.com/libsdl-org/SDL/blob/main/include/SDL3/SDL_events.h#L645-L665)? In your tests you were using Xbox game controllers, so I'm a bit confused why these report themselves as "joysticks", not "game controllers". What about the extra buttons on game controllers like the Xbox 360; those are fully mappable in Staging. So wouldn't it be better if we only dealt with "game controllers" conceptually?

I did not get game controller events in my testing. In fact, I tried those events first. FWICT there is no particular reason to exclude them though, nor is the logic different; so I've added cases for them that do the same thing. My guess is:
* controller that has any axes at all -> joystick
* controller with only buttons (and maybe hats) -> controller

But just a guess, and I don't have any peripherals that trigger controller events instead of joystick events.

> So, I suggest to log something like this instead: MAPPER: Game controller Xbox 360 has been plugged in." (or whatever the human-readable name of the controller is). And then MAPPER: Game controller Xbox 360 has been disconnected."

Changed for now to "Game controller X has been plugged in" until I can find out how to look up human friendly name at runtime.

Reply:
OK, now the human-readable joystick name is used. Here is some sample output:
```
2024-12-27 01:45:18.811 | MAPPER: Game controller 'Xbox 360 Wireless Controller' has been disconnected
2024-12-27 01:45:21.916 | MAPPER: Game controller 'Xbox 360 Wireless Controller' has been plugged in
```

Reply:
Thanks for addressing my comments; those human-readable log message look perfect now. Yeah maybe let's just stick with the joystick concept for now given that the mapper seems to mention "joysticks" all over the place. We'll just need to test things a bit to make sure the buttons on gamepads are still mappable (even though SDL insists on calling them "joysticks", which is a bit weird... in my view, there are only controllers, and that's it).

I'll admit I use controllers with Staging very very rarely, but no matter, we'll make sure with the team that things work as intended.